### DISCUSSION TITLE
what tools should accept blacklisted regions and in what form?

### DISCUSSION THREAD

**Discussion by sergpolly** [open2c/cooltools#286] [Ideas]
I was trying to fill some placeholders in CLI expected (and expected.py module itself) and was looking for a way to read/parse a bed-file with "bad bins" - here is a candidate:
https://github.com/mirnylab/cooler/blob/c9c718fdebccbda41ad10c47f700853f79ee3cd3/cooler/cli/balance.py#L175

although there are some edge-case issues https://github.com/mirnylab/cooler/issues/196 and the problem seems very general enough - maybe there is something in `bioframes` that I'm not aware of ?

**Comment by golobor** [comment]
my personal quick thought on this - blacklisting should only be done at stage of balancing and this information would then be reused by downstream tools via the proper choice of a balancing weight column
